reactive_stores/src/option.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{StoreField, Subfield};
+use reactive_graph::traits::Read;
+use std::ops::Deref;
+
+pub trait OptionStoreExt
+where
+    Self: StoreField<Value = Option<Self::Output>>,
+{
+    type Output;
+
+    fn unwrap(self) -> Subfield<Self, Option<Self::Output>, Self::Output>;
+
+    fn map<U>(
+        self,
+        map_fn: impl FnOnce(Subfield<Self, Option<Self::Output>, Self::Output>) -> U,
+    ) -> Option<U>;
+}
+
+impl<T, S> OptionStoreExt for S
+where
+    S: StoreField<Value = Option<T>> + Read,
+    <S as Read>::Value: Deref<Target = Option<T>>,
+{
+    type Output = T;
+
+    fn unwrap(self) -> Subfield<Self, Option<Self::Output>, Self::Output> {
+        Subfield::new(
+            self,
+            0.into(),
+            |t| t.as_ref().unwrap(),
+            |t| t.as_mut().unwrap(),
```

Comment:
I got a panic here on line 31 more than once (I think on page reload), but not reliably reproducible, unwrapping a `None` value.

Reply:
The problem actually was, I called `.set` on the subfield value inside the closure, and this broke after that. Works if I call `.set(Some(...))` on the value outside the closure.

Reply:
Do you have a quick example? I think I understand what you mean but am not totally sure.

Reply:
Here it is - the subfield coming out of `map` is stale unless the parent field is triggered.

Click `Button`, and then `Button 2`. Bug or PEBKAC?

```rust
#[derive(Debug, Clone, reactive_stores_macro::Store)]
pub struct User {
    pub name: Option<Name>,
}

#[derive(Debug, Clone, reactive_stores_macro::Store)]
pub struct Name {
    pub first_name: Option<&'static str>,
}

#[component]
fn App() -> impl IntoView {
    let store = Store::new(User {
        name: Some(Name {
            first_name: Some("John Doe"),
        }),
    });
    view! {
        {move || {
            store
                .name()
                .map({
                    move |name| {
                        view! {
                            <button on:click=move |_| {
                                name.set(Name { first_name: None });
                            }>Button</button>
                            <br />
                            {move || {
                                name.first_name()
                                    .map({
                                        move |name| {
                                            view! {
                                                <button on:click=move |_| {
                                                    name.set("James Doe")
                                                }>Button 2</button>
                                            }
                                        }
                                    })
                            }}
                        }
                    }
                })
        }}
    }
}
```

Reply:
Ah ok interesting. This is a great test case for whether accessing a child field should also track the parent, which was the change [here](https://github.com/leptos-rs/leptos/pull/2871/commits/fb9347923138741ff51a946b668f0adcfef615cf).

The reasoning to *not* track them was that it over-re-runs some things: for example, changing from `Some("John Doe")` to `Some("James Doe")` would cause the `.map()`. 

But I think this is an obvious reason that change was wrong.

Reply:
Ok, I actually read your example too quickly: What's *really* going on here is a simpler issue, namely that reading from a subfield was not appropriately tracking its parents. What this meant was that mutating `first_name` by mutating `name` was not notifying the effect that read from `first_name`, which is clearly wrong.

97282f4 is the one-line fix, and thanks again for pointing this out.